gtsfm/two_view_estimator.py

```diff
@@ -26,20 +27,67 @@
 pil_logger.setLevel(logging.INFO)
 
 
+# In case an epipolar geometry can be verified, it is checked whether
+# the geometry describes a planar scene or panoramic view (pure rotation)
+# described by a homography. This is a degenerate case, since epipolar
+# geometry is only defined for a moving camera. If the inlier ratio of
+# a homography comes close to the inlier ratio of the epipolar geometry,
+# a planar or panoramic configuration is assumed.
+# Based on COLMAP's front-end logic here:
+#    https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L230
+MAX_H_INLIER_RATIO = 0.8
+
+EPSILON = 1e-6
+
+
+@dataclass(frozen=False)
+class TwoViewEstimationReport:
+    """
+    Args:
+        num_inliers_est_model: #correspondences consistent with estimated model (not necessarily "correct")
+        inlier_ratio_est_model: measures how polluted the putative matches were
+        num_inliers_gt_model: measures how well the verification worked, w.r.t. GT
+        inlier_ratio_gt_model: Only defined if GT relative pose provided
+        R_error_deg: relative pose error. Only defined if GT poses provided
+        U_error_deg
```

Comment:
Missing explanation.

Reply:
thanks -- docstring should be cleaned up now